lib/std/elf.zig

```diff
@@ -387,8 +388,14 @@ pub const Header = struct {
             else => return error.InvalidElfClass,
         };
 
+        const machine = if (need_bswap) blk: {
+            const value = @enumToInt(hdr32.e_machine);
```

Comment:
The use of `ptrCast` is dangerous, here you have a `hdr32.e_machine` with an invalid value due to the reversed byte ordering (and `EM` being exhaustive).
The `intToEnum` involuntarily adds a layer of protection, but only in this code path.

I'd suggest to turn `ET` into non-exhaustive enums to prevent any further problem.

Reply:
Makes sense. (This is what I was alluding to in the PR body.)

Fixed in 24e5959.